java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java

```diff
@@ -179,6 +183,117 @@ public static int getNullCount(final ArrowBuf validityBuffer, final int valueCou
     return 8 * sizeInBytes - count;
   }
 
+  /**
+   * Tests if all bits in a validity buffer are 0.
+   * @param validityBuffer the validity buffer.
+   * @param valueCount the bit count.
+   * @return true if all bits are 0, and false otherwise.
+   */
+  public static boolean allBitsNull(final ArrowBuf validityBuffer, final int valueCount) {
+    if (valueCount == 0) {
+      return true;
+    }
+    final int sizeInBytes = getValidityBufferSize(valueCount);
+
+    // boundary check
+    validityBuffer.checkBytes(0, sizeInBytes);
+
+    // If value count is not a multiple of 8, then calculate number of used bits in the last byte
+    final int remainder = valueCount % 8;
+    final int fullBytesCount = remainder == 0 ? sizeInBytes : sizeInBytes - 1;
+
+    int index = 0;
+    while (index + 8 <= fullBytesCount) {
+      long longValue = getLong(validityBuffer.memoryAddress() + index);
+      if (longValue != 0L) {
+        return false;
+      }
+      index += 8;
+    }
+
+    while (index + 4 <= fullBytesCount) {
+      int intValue = getInt(validityBuffer.memoryAddress() + index);
+      if (intValue != 0) {
+        return false;
+      }
+      index += 4;
+    }
+
+    while (index < fullBytesCount) {
+      byte byteValue = getByte(validityBuffer.memoryAddress() + index);
+      if (byteValue != (byte) 0) {
+        return false;
+      }
+      index += 1;
+    }
+
+    // handling with the last bits
+    if (remainder != 0) {
+      byte byteValue = getByte(validityBuffer.memoryAddress() + sizeInBytes - 1);
+      byte mask = (byte) ((1 << remainder) - 1);
+      byteValue = (byte) (byteValue & mask);
+      if (byteValue != (byte) 0) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Tests if all bits in a validity buffer are 1.
+   * @param validityBuffer the validity buffer.
+   * @param valueCount the bit count.
+   * @return true if all bits are 1, and false otherwise.
+   */
+  public static boolean allBitsSet(final ArrowBuf validityBuffer, final int valueCount) {
+    if (valueCount == 0) {
+      return true;
+    }
+    final int sizeInBytes = getValidityBufferSize(valueCount);
+
+    // boundary check
+    validityBuffer.checkBytes(0, sizeInBytes);
+
+    // If value count is not a multiple of 8, then calculate number of used bits in the last byte
+    final int remainder = valueCount % 8;
+    final int fullBytesCount = remainder == 0 ? sizeInBytes : sizeInBytes - 1;
+
+    int index = 0;
+    while (index + 8 <= fullBytesCount) {
```

Comment:
can we merge this with the code in above method..the only thing to change looks like the comparison value 0 and 1?

Reply:
@praveenbingo Greate suggestion. Thank you.